pkg/admission/validator/shoot.go

```diff
@@ -20,28 +20,36 @@ import (
 
 	extensionswebhook "github.com/gardener/gardener/extensions/pkg/webhook"
 	"github.com/gardener/gardener/pkg/apis/core"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 
 	api "github.com/gardener/gardener-extension-registry-cache/pkg/apis/registry"
+	"github.com/gardener/gardener-extension-registry-cache/pkg/apis/registry/v1alpha1/helper"
 	"github.com/gardener/gardener-extension-registry-cache/pkg/apis/registry/validation"
 )
 
 // shoot validates shoots
 type shoot struct {
-	decoder runtime.Decoder
+	apiReader client.Reader
+	decoder   runtime.Decoder
 }
 
 // NewShootValidator returns a new instance of a shoot validator.
-func NewShootValidator(decoder runtime.Decoder) extensionswebhook.Validator {
+func NewShootValidator(mgr manager.Manager) extensionswebhook.Validator {
```

Comment:
Don't pass `mgr manager.Manager` but pass only what it is required to the validator. The fact that someone did it like this for the other extensions doesn't mean that it is the correct way. 

```suggestion
func NewShootValidator(apiReader client.Reader, decoder runtime.Decoder) extensionswebhook.Validator {
```

This improves the unit testing as you need to pass to the validator only the things it needs. Otherwise you have to pass it a manager.Manager and the validator only uses 2 fields from the whole manager.

After adapting `vendor/github.com/gardener/gardener/pkg/mock/controller-runtime/manager` should be dropped as it is no longer needed.